pkg/operator/apis/monitoring/v1/pod_config.go

```diff
@@ -382,6 +386,76 @@ func relabelingsForMetadata(keys map[string]struct{}) (res []*relabel.Config) {
 			TargetLabel:  "node",
 		})
 	}
+	if _, ok := keys["top_level_controller_name"]; ok {
+		res = append(res,
+			// Default to pod name
+			&relabel.Config{
+				Action:       relabel.Replace,
+				SourceLabels: prommodel.LabelNames{"__meta_kubernetes_pod_name"},
+				TargetLabel:  "top_level_controller_name",
+			},
+			// For non-ReplicaSet and non-Job controllers
+			&relabel.Config{
+				Action:       relabel.Replace,
+				SourceLabels: prommodel.LabelNames{"__meta_kubernetes_pod_controller_name"},
+				Regex:        relabel.MustNewRegexp("(.+)"),
```

Comment:
Do we need this (and `Replacement`) in this entry?

That seems to be [a Prometheus default anyway](https://github.com/prometheus/prometheus/blob/main/model/relabel/relabel.go#L32)?